General Cleanup For Cocoapods

twof opened the pull request:
- Added a license
- Bumped Swift version to 4.2
- Bumped iOS release version from 10.1 to 10.3
- All warnings silenced for cocoapods linter except from the tag requirement warning which will be fixed upon merge and tag

alan-yilun-li committed:
updated for redbooth

twof committed:
Updated to swift 4

alan-yilun-li committed:
Merge pull request #1 from redbooth/swift4

Updated fork to Swift 4

rahuljiresal committed:
made changes to add xcode10 and swift 4.2 support

rahuljiresal committed:
renamed a variable

rahuljiresal committed:
Merge pull request #2 from redbooth/fix/xcode10

made changes to add xcode10 and swift 4.2 support

rahuljiresal committed:
Merge branch 'om' into fix/iOS12

rahuljiresal committed:
Merge pull request #3 from redbooth/fix/iOS12

Fix/ios12

twof committed:
removed Redbooth remanents

twof committed:
Merge pull request #4 from redbooth/prepForRelease

removed Redbooth remanents

twof committed:
cleanup for cocoapods release

twof committed:
Merge pull request #5 from redbooth/prepForRelease

cleanup for cocoapods release

twof committed:
update podspec to redboothImagePicker

twof committed:
podspec passing

radimhalfar commented on ImagePicker/AVPreviewView.swift:
```diff
@@ -24,51 +24,51 @@ enum VideoDisplayMode {
-        case .aspectFit:    previewLayer.videoGravity = AVLayerVideoGravity.resizeAspect
-        case .resize:       previewLayer.videoGravity = AVLayerVideoGravity.resize
+
+    private func applyVideoDisplayMode() {        switch displayMode {
```
fix alignment

radimhalfar commented on ImagePicker/ActionCell.swift:
```diff
@@ -9,30 +9,38 @@
-    @IBOutlet weak var imageView: UIImageView!
-    
+public final class ActionCell : UICollectionViewCell {
+
```
don't use extra opening lines

radimhalfar commented on ImagePicker/ActionCell.swift:
```diff
@@ -9,30 +9,38 @@
-    
+
+    public func setupOffsets() {
+        topOffset.constant = 5
```
DRY, keep 5 to a constant

radimhalfar commented on ImagePicker/AssetCell.swift:
```diff
@@ -28,43 +28,43 @@ public protocol ImagePickerAssetCell : class {
+
         durationLabel.textColor = UIColor.white
-        durationLabel.font = UIFont.systemFont(ofSize: 12, weight: .semibold)
+        durationLabel.font = UIFont.systemFont(ofSize: 12, weight: UIFont.Weight.semibold)
```
keep .semibold

radimhalfar commented on ImagePicker/CarvedLabel.swift:
```diff
@@ -24,84 +24,84 @@ final class CarvedLabel : UIView {
+
     fileprivate var textAttributes: TextAttributes {
-        let activeFont = font ?? UIFont.systemFont(ofSize: 12, weight: .regular)
+        let activeFont = font ?? UIFont.systemFont(ofSize: 12, weight: UIFont.Weight.regular)
```
keep .regular

radimhalfar commented on ImagePicker/ImagePickerController.swift:
```diff
@@ -199,67 +220,68 @@ open class ImagePickerController : UIViewController {
+    fileprivate var collectionViewCoordinator: CollectionViewUpdatesCoordinator!
+
     fileprivate var imagePickerView: ImagePickerView! {
-        return view as! ImagePickerView
```
this doesn't affects functionality. if kept as fatal please specify error message if possible

radimhalfar commented on ImagePicker/ImagePickerController.swift:
```diff
@@ -353,118 +375,118 @@ open class ImagePickerController : UIViewController {
+
         //perform update animations
-        collectionViewCoordinator.performChanges(changes, inSection: layoutConfiguration.sectionIndexForAssets)
+        collectionViewCoordinator.performChanges(changes as! PHFetchResultChangeDetails<PHObject>, inSection: layoutConfiguration.sectionIndexForAssets)
```
don't use force casts, rather assert it or guard it

radimhalfar commented on ImagePicker/ImagePickerController.swift:
```diff
@@ -474,46 +496,46 @@ extension ImagePickerController : ImagePickerDelegateDelegate {
+
         //update authorization status if it's changed
-        let status = AVCaptureDevice.authorizationStatus(for: .video)
+        let status = AVCaptureDevice.authorizationStatus(for: AVMediaType.video)
```
keep .video only

radimhalfar commented on ImagePicker/NoPermissionsView.swift:
```diff
@@ -0,0 +1,83 @@
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        label.font = .systemFont(ofSize: 17, weight: UIFont.Weight.regular)
```
keep .regular

radimhalfar commented on ImagePicker/NoPermissionsView.swift:
```diff
@@ -0,0 +1,83 @@
+        let button = UIButton(type: .system)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setTitle(NSLocalizedString("Go to Settings", comment: ""), for: .normal)
+        button.titleLabel?.font = .systemFont(ofSize: 20, weight: UIFont.Weight.semibold)
```
.semibold

radimhalfar commented on ImagePicker/PhotoCaptureDelegate.swift:
```diff
@@ -1,154 +1,155 @@
-//    }
-    
+    //    @available(iOS 11.0, *)
+    //    func photoOutput(_ output: AVCapturePhotoOutput, didFinishProcessingPhoto photo: AVCapturePhoto, error: Error?) {
```
remove commented out code

radimhalfar reviewed (changes requested):
👏 Look good, please update minors and we can merge it

twof commented:
Thanks for your feedback! I no longer have edit access at the Redbooth org, so @lollar or Eli is going to have to fix this up. (Mike can you tag Eli? I don't remember his handle. Also feel free to reach out to me if you want help fixing this up.)

xavisavvy committed:
- Updated to Swift 5
- Updated all the plist settings w/ any new configs
- Removed unnecessary / extra public function calls

xavisavvy committed:
Updated gitignore, and podspec version

xavisavvy committed:
Merge pull request #6 from redbooth/swift-5-upgrade

Swift 5 upgrade

xavisavvy committed:
Added XCode 12 fixes for unknown returns in switches.

fernandogcat committed:
Add Package.swift for SPM support
